python/lib/dependabot/python/update_checker/pip_compile_version_resolver.rb

```diff
@@ -235,35 +235,6 @@ def pip_compile_index_options
         def run_pip_compile_command(command)
           run_command("pyenv local #{python_version}")
           run_command(command)
-        rescue SharedHelpers::HelperSubprocessFailed => e
-          original_err ||= e
-          msg = e.message
-
-          relevant_error = choose_relevant_error(original_err, e)
-          raise relevant_error unless error_suggests_bad_python_version?(msg)
-          raise relevant_error if user_specified_python_version
-          raise relevant_error if python_version == "2.7.18"
-
-          @python_version = "2.7.18"
-          retry
-        ensure
-          @python_version = nil
-          FileUtils.remove_entry(".python-version", true)
-        end
-
-        def choose_relevant_error(previous_error, new_error)
-          return previous_error if previous_error == new_error
-
-          # If the previous error was definitely due to using the wrong Python
-          # version, return the new error (which can't be worse)
-          return new_error if error_certainly_bad_python_version?(previous_error.message)
-
-          # Otherwise, if the new error may be due to using the wrong Python
-          # version, return the old error (which can't be worse)
-          return previous_error if error_suggests_bad_python_version?(new_error.message)
-
-          # Otherwise, default to the new error
-          new_error
```

Comment:
Same here, retry is ran when an update failed on python 3 and retries with python 2, this is now no longer required